Add table of known identity values

Pennycook opened the pull request:
Clarifies which combinations of BinaryOperator and AccumulatorT are
guaranteed to have known identities, and the values of those identities.

Closes #163.

psalz commented on adoc/chapters/programming_interface.adoc:
```diff
@@ -12525,6 +12525,182 @@ be tested using the [code]#has_known_identity# trait class.
+----
+a@
+----
+std::numeric_limits<AccumulatorT>::infinity()
```
As pointed out by @fknorr in #163, the spec currently does not mandate that `std::numeric_limits` be specialized for `sycl::half`.

keryell replied:
> As pointed out by @fknorr in #163, the spec currently does not mandate that `std::numeric_limits` be specialized for `sycl::half`.

Perhaps we should?

gmlueck commented on adoc/chapters/programming_interface.adoc:
```diff
@@ -12525,6 +12525,182 @@ be tested using the [code]#has_known_identity# trait class.
 include::{header_dir}/identity.h[lines=4..-1]
```
I wanted to  make this comment on the previous paragraph that starts "Certain implementations of reductions can be made more efficient ...", but GitHub won't let me comment on an unchanged paragraph.

I wonder if the intent would be more clear if this paragraph started out like:

> Some of the overloads for the `reduction` function take an identity value and some do not.  An implementation is required to compute a correct reduction even when the application does not specify an identity value.  However, the implementation may be more efficient when the identity value is either provided by the application or is known by the implementation.  [...]

Pennycook replied:
I like this.  I've made the change in 954933ce3a9411fa7f21170803ff31d3da2b6622.  Since I was updating this paragraph, I also folded in @psalz's feedback from https://github.com/KhronosGroup/SYCL-Docs/issues/163#issuecomment-1240678038 to clarify that the identity depends on more than the binary operator alone.

keryell reviewed (approved):
Good clarification.

fraggamuffin commented:
should we fix numeric_limits for half now? (will require impl to change)or do it for SYCL NEXT?
Seems ok now.

fraggamuffin commented:
Can we also say all std traits work?

fraggamuffin commented:
http://eel.is/c++draft/numeric.limits.general#6

fraggamuffin commented:
Non-arithmetic standard types, such as [complex<T>](http://eel.is/c++draft/complex), shall not have specializations[.](http://eel.is/c++draft/numeric.limits.general#6.sentence-1)

Pennycook commented:
I've added the wording about `std::numeric_limits<sycl::half>`, but in a different place to where #163 suggested.

#163 suggested that it should be added to the [Half precision floating-point](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:opencl:extension-fp16) section,  but this is a subsection of the specification related to OpenCL.  The `std::numeric_limits` requirement should apply to all backends, so I've added the requirement to the table that first defines `sycl::half`.

Pennycook commented:
> Can we also say all std traits work?

I now think we should defer this to a later PR, so we can spend more time getting the language right.

The specification already makes reference to `std::numeric_limits<sycl::half>` in device code for some math functions (via `std::numeric_limits<genfloat>`, so this PR isn't introducing any new requirements, just clarifying them.
